app/src/main/java/com/voltix/wallet/service/CryptoPriceService.kt

```diff
@@ -0,0 +1,92 @@
+package com.voltix.wallet.service
+
+import com.google.gson.Gson
+import com.voltix.wallet.common.Endpoints
+import com.voltix.wallet.common.SettingsCurrency
+import com.voltix.wallet.data.common.data_store.AppDataStore
+import com.voltix.wallet.models.Coin
+import com.voltix.wallet.models.Coins
+import com.voltix.wallet.models.CryptoPrice
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.URL
+import java.util.Date
+import javax.inject.Inject
+
+class CryptoPriceService @Inject constructor(
+    private val appDataStore: AppDataStore
+) {
+    private val cache: MutableMap<String, Pair<CryptoPrice, Date>> = mutableMapOf()
```

Comment:
The service can be used on different thread , and MutableMap is not threadsafe , could you please use Guava cache instead?